src/createBox.ts

```diff
@@ -20,36 +21,49 @@ import {
   PositionProps,
   visible,
   VisibleProps,
+  SpacingShorthandProps,
+  BackgroundColorShorthandProps,
+  spacingShorthand,
 } from './restyleFunctions';
 
-export type BoxProps<Theme extends BaseTheme> = BackgroundColorProps<Theme> &
+export type BoxProps<
+  Theme extends BaseTheme,
+  EnableShorthand extends boolean = true
+> = BackgroundColorProps<Theme> &
   OpacityProps<Theme> &
   VisibleProps<Theme> &
   LayoutProps<Theme> &
   SpacingProps<Theme> &
   BorderProps<Theme> &
   ShadowProps<Theme> &
-  PositionProps<Theme>;
+  PositionProps<Theme> &
+  (EnableShorthand extends true
+    ? SpacingShorthandProps<Theme> & BackgroundColorShorthandProps<Theme>
+    : never);
```

Comment:
This seems to be causing problems, see: 

![image](https://user-images.githubusercontent.com/688415/88179932-e0c73380-cc2c-11ea-96c2-35786ca62d23.png)

Replacing `never` here with `Record<string, never>` seems to work well though.